src/types/checkType.js

```diff
@@ -11,6 +11,11 @@ const recursiveCheck = (response, type) => {
     return response.reduce((a, b) => a && recursiveCheck(b, type.items), true);
   }
   if (type.type === "object") {
+    const isOptionalAndNull = type.type.optional && response === null;
```

Comment:
Ok, what I don't like about this solution is, that this actually does not work...

Take a look at this:

```js
const schema = obj({
  number: int().optional()
})

const value = { number: null };
```

With the current logic, this would not match, as you only check within the object type.

The way it should be implemented is, by checking all of an objects children (as only within an object, the concept of optional keys exist anyways).

Line 24 is the culprit here:
The way it works for now is, that only the **existing** keys on the object are checked `Object.keys(response)`. But a field, that is null does exist. So what you could do would be to filter the keys for null-values and also modify line 29 to allow null as non-existing.